Development/nmos/settings.h

```diff
@@ -359,6 +366,86 @@ namespace nmos
+            const web::json::field_as_value authorization_services{ U("authorization_services") };
+
+            // authorization_request_max [registry, node]: timeout for interactions with the Authorization API /certs & /token endpoints
+            const web::json::field_as_integer_or authorization_request_max{ U("authorization_request_max"), 30 };
+
+            // fetch_authorization_public_keys_interval_min/fetch_authorization_public_keys_interval_max [registry, node]: used to poll for Authorization API public keys changes; default is about one hour
+            // "Resource Servers (Nodes) SHOULD seek to fetch public keys from the Authorization Server at least once every hour. Resource Servers MUST vary their retrieval
+            // interval at random by up to at least one minute to avoid overloading the Authorization Server due to Resource Servers synchronising their retrieval time."
+            // See https://specs.amwa.tv/is-10/releases/v1.0.0/docs/4.1._Behaviour_-_Authorization_Servers.html#authorization-server-public-keys
+            const web::json::field_as_integer_or fetch_authorization_public_keys_interval_min{ U("fetch_authorization_public_keys_interval_min"), 3600 };
+            const web::json::field_as_integer_or fetch_authorization_public_keys_interval_max{ U("fetch_authorization_public_keys_interval_max"), 3660 };
+
+            // access_token_refresh_interval [node]: time interval (in seconds) to refresh access token from Authorization Server
+            // It specified the access token refresh period otherwise Bearer token's expires_in is used instead.
+            // See https://specs.amwa.tv/is-10/releases/v1.0.0/docs/4.4._Behaviour_-_Access_Tokens.html#access-token-lifetime
+            const web::json::field_as_integer_or access_token_refresh_interval{ U("access_token_refresh_interval"), -1 };
+
+            // client_authorization [node]: whether clients should use authorization to access protected APIs
+            const web::json::field_as_bool_or client_authorization{ U("client_authorization"), false };
+
+            // server_authorization [registry, node]: whether server should use authorization to protect its APIs
+            const web::json::field_as_bool_or server_authorization{ U("server_authorization"), false };
+
+            // authorization_code_flow_max [node]: timeout for the authorization code workflow (in seconds)
+            // No timeout if value is set to -1, default to 30 seconds
+            const web::json::field_as_integer_or authorization_code_flow_max{ U("authorization_code_flow_max"), 30 };
+
+            // authorization_flow [node]: used to specify the authorization flow for the registered scopes
+            // supported flow are authorization_code and client_credentials
+            // client_credentials SHOULD only be used for NO user interface node/registry, otherwise authorization_code MUST be used
```

Comment:
```suggestion
            // client_credentials SHOULD only be used when the node/registry has NO user interface, otherwise authorization_code MUST be used
```